### PR TITLE
Implement threat

### PR DESCRIPTION
**Pull request by jokal2**

### CODE REVIEW

**Comment by jokal2** [comment]
Still have to play with it to figure out what all needs to be done. But I figured I'd put it up as a RFC.

File: Statistics/StatisticsSpell.cpp [hunk 1]
```diff
@@ -324,6 +324,55 @@ void StatisticsSpell::add_dmg(const Outcome outcome, const int dmg, const double
...
+    if (threat_per_resource < min_tpr)
+        min_tpr = threat_per_resource;
+    if (threat_per_resource > max_tpr)
+        max_tpr = threat_per_resource;
+
+    avg_tpr = avg_tpr + (threat_per_resource - avg_tpr) / threat_dealt_successes;
+}
+
+void StatisticsSpell::add_tpet(const int thrt, const double execution_time) {
+    if (delta(execution_time, 0) < 0.0001)
```
Comment (timhul) [inline Statistics/StatisticsSpell.cpp]: Use `almost_equal` here as well?

**Comment by jokal2** [comment]
This is still WIP -- it generates statistics for threat but some of the UI bindings seem off. I'm basically just guessing on the Qt stuff and grepping for words.

Unfortunately this is creating a ton of copy/paste boilerplate type code. A better overall idea might be to attach

**Comment by timhul** [comment]
I don't see anything particularly wrong in the UI, what were you referring to?

Let me know if you need help or when you feel this is ready for merging.

**Comment by jokal2** [comment]
`Tot. Thrt %` and `Max Hit` and `Max Crit` are off in the GUI. I can't find anything that would cause this. I don't see anything else wrong, though.

**Comment by timhul** [comment]
Good job

**Pull request merged**